packages/grid/_modules_/grid/components/menu/GridMenu.tsx

```diff
@@ -39,14 +39,24 @@ export const GridMenu: React.FC<MenuProps> = ({
   position,
   ...other
 }) => {
+  const targetRef = React.useRef(target);
+  const openRef = React.useRef(open);
+
+  React.useEffect(() => {
+    if (openRef.current && targetRef.current) {
+      (targetRef.current as HTMLElement).focus();
+    }
+
+    openRef.current = open;
+    targetRef.current = target;
+  }, [open, target]);
```

Comment:
This looks wrong. I can still see the focus move to the body, then to the action. It should work by default. If it isn't and we need a workaround, we need to know why, open a detailed issue in the main repository so the patch stays temporary. I would also try removing GridMenu and use the Menu component directly. It would incentive/focus us to solve the issue in the main repo and not workaround it, delivering a fraction of the value that we could possibly do by solving it for everybody.

Reply:
Ok, I tried it with the `Menu` component and it does work but we are using this implementation from the docs https://material-ui.com/components/menus/#menulist-composition and it there is basically the same logic of focusing the anchor element on close.
I don't think there is a bug with either approach it's just that we chose to do it this way and we need to handle the refocus.

Reply:
I don't understand. If https://material-ui.com/components/menus/#menulist-composition is working, and we are using the same approach, then it should be working.

Reply:
I had a closer look. It seems that I was confused. We weren't following https://material-ui.com/components/menus/#menulist-composition before. Now, we are, all good.